contracts/EVMCode.slb

```diff
@@ -0,0 +1,67 @@
+pragma solidity 0.5.2;
+pragma experimental ABIEncoderV2;
+
+
+library EVMCode {
+
+    struct Code {
+        address codeAddress;
+        bytes codeBytes;
+        uint length;
+    }
+
+    function fromAddress(address codeAddress) internal view returns (Code memory code) {
+        code.codeAddress = codeAddress;
+        uint codeSize;
+        assembly {
+            codeSize := extcodesize(codeAddress)
+        }
+
+        code.length = codeSize;
+    }
+
+    function fromBytes(bytes memory codeBytes) internal pure returns (Code memory code) {
+        code.codeBytes = codeBytes;
+        code.length = codeBytes.length;
+    }
+
+    function getOpcodeAt(Code memory self, uint pos) internal view returns (uint8 opcode) {
+        address codeContractAddress = self.codeAddress;
+
+        if (codeContractAddress == address(0)) {
+            opcode = uint8(self.codeBytes[pos]);
+        } else {
+            assembly {
+                extcodecopy(codeContractAddress, 31, pos, 1)
+                opcode := mload(0)
+            }
+        }
+    }
+
+    function toBytes(Code memory self) internal view returns (bytes memory bts) {
+        address codeContractAddress = self.codeAddress;
+
+        if (codeContractAddress == address(0)) {
+            bts = self.codeBytes;
+        } else {
+            uint size = self.length;
+            assembly {
+                bts := mload(0x40)
+                mstore(0x40, add(bts, and(add(add(size, 0x20), 0x1f), not(0x1f))))
```

Comment:
please help me understand this line :smile:
I see you are allocate memory for code
what is the meaning of the `and( ... )` part?
And I think this line is worthy for some comment :thought_balloon:

Reply:
Ya, that is padding up to a whole word